### ISSUE TITLE
Release Version 2.2.0

### ISSUE THREAD

**Issue by peterzhuamazon** [opensearch-project/k-NN#467]
This is a component issue for __2.2.0__.
Coming from [opensearch-build#2271](https://github.com/opensearch-project/opensearch-build/issues/2271). Please follow the following checklist.
__Please refer to the DATES in that post__.

<details><summary>How to use this issue</summary>
<p>

## This Component Release Issue

This issue captures the state of the OpenSearch release, on component/plugin level; its assignee is responsible for driving the release. Please contact them or @mention them on this issue for help.
Any release related work can be linked to this issue or added as comments to create visiblity into the release status.

## Release Steps

There are several steps to the release process; these steps are completed as the whole component release and components that are behind present risk to the release. The component owner resolves the tasks in this issue and communicate with the overall release owner to make sure each component are moving along as expected.

Steps have completion dates for coordinating efforts between the components of a release; components can start as soon as they are ready far in advance of a future release. The most current set of dates is on the overall release issue linked at the top of this issue.

## The Overall Release Issue

Linked at the top of this issue, the overall release issue captures the state of the entire OpenSearch release including references to this issue, the release owner which is the assignee is responsible for communicating the release status broadly.  Please contact them or @mention them on that issue for help.

## What should I do if my plugin isn't making any changes?

If including changes in this release, increment the version on `2.0` branch to `2.2.0` for Min/Core, and `2.2.0.0` for components. Otherwise, keep the version number unchanged for both.

</p>
</details>

### Preparation

- [x] Assign this issue to a release owner.
- [x] Finalize scope and feature set and update [the Public Roadmap](https://github.com/orgs/opensearch-project/projects/1).
- [x] All the tasks in this issue have been reviewed by the release owner.
- [x] Create, update, triage and label all features and issues targeted for this release with `v2.2.0`.
- [x] Cut 2.2 branch

### CI/CD

- [x] All code changes for `2.2.0` are complete.
- [x] Ensure working and passing CI.
- [x] Check that this repo is included in the [distribution manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.2.0).

### Pre-Release

- [x] Update to the `2.2.0` release branch in the [distribution manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.2.0).
- [ ] Increment the version on the parent branch to the next development iteration.
- [x] Gather, review and publish release notes following the [rules](https://github.com/opensearch-project/opensearch-plugins/blob/main/RELEASE_NOTES.md) and back port it to the release branch.[git-release-notes](https://github.com/ariatemplates/git-release-notes) may be used to generate release notes from your commit history.
- [x] Confirm that all changes for `2.2.0` have been merged.
- [ ] Add this repo to the [manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/) for the next developer iteration.

### Release Testing

- [x] Find/fix bugs using latest tarball and docker image provided in parent release issue and update the release notes if necessary.
- [x] Code Complete: Test within the distribution, ensuring integration, backwards compatibility, and performance tests pass.
- [x] Sanity Testing: Sanity testing *and* fixing of critical issues found.
- [x] File issues for all intermittent test failures.

### Release

- [x] Complete [documentation](https://github.com/opensearch-project/documentation-website).
- [x] Verify all issued labeled for this release are closed or labelled for the next release.

### Post Release

- [x] Prepare for an eventual security fix development iteration by incrementing the version on the release branch to the next eventual patch version.
- [x] Add this repo to the [manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/) of the next eventual security patch version.
- [x] [Suggest improvements](https://github.com/opensearch-project/opensearch-build/issues/new) to [this template](https://github.com/opensearch-project/opensearch-build/blob/main/meta/templates/releases/release_template.md).
- [x] Conduct a retrospective, and publish its results.

**Comment by junqiu-lei** [comment]
Executed benchmark script part of [k-NN perf-tool](https://github.com/opensearch-project/k-NN/tree/main/benchmarks/perf-tool) with [nmslib-sift-hnsw/test.yml](https://github.com/opensearch-project/k-NN/blob/main/benchmarks/perf-tool/sample-configs/nmslib-sift-hnsw/test.yml):
```
python knn-perf-tool.py test /home/ec2-user/k-NN/benchmarks/perf-tool/sample-configs/nmslib-sift-hnsw/test.yml output.json
```
Results
```
INFO:root:Finished running tests.
{
  "metadata": {
    "test_name": "nmslib_sift_hnsw",
    "test_id": "Test workflow for nmslib hnsw",
    "date": "08/10/2022 23:27:33",
    "python_version": "3.8.13 (default, Mar 28 2022, 11:38:47) \n[GCC 7.5.0]",
    "os_version": "Linux-5.10.130-118.517.amzn2.x86_64-x86_64-with-glibc2.17",
    "processor": "x86_64, 36 cores",
    "memory": "737492992 (used) / 72107167744 (available) / 73638821888 (total)"
  },
  "results": {
    "test_took": 1781505.458917499,
    "create_index_took_total": 256.23650300076406,
    "ingest_took_total": 85910.8443339992,
    "refresh_index_store_kb_total": 1501464.1430664062,
    "refresh_index_took_total": 1615729.6513039982,
    "force_merge_took_total": 4327.72677650064,
    "query_took_total": 75281.0,
    "query_took_p50": 7.5,
    "query_took_p90": 8.5,
    "query_took_p99": 10.0,
    "query_memory_kb_total": 652864.5,
    "query_recall@K_total": 0.9983885,
    "query_recall@1_total": 1.0
  }
```

**Comment by junqiu-lei** [comment]
Executed benchmark script part of k-NN osb (https://github.com/opensearch-project/k-NN/tree/main/benchmarks/osb):
```
export PARAMS_FILE=/home/ec2-user/k-NN/benchmarks/osb/params/no-train-params.json
export PROCEDURE=no-train-test
opensearch-benchmark execute_test --target-hosts $URL:$PORT --workload-path ./workload.json --workload-params ${PARAMS_FILE} --test-procedure=${PROCEDURE} --pipeline benchmark-only
```

```
[INFO] Executing test with workload [workload], test_procedure [no-train-test] and provision_config_instance [‘external’] with version [2.2.0].
Running delete-target-index                                                    [100% done]
Running create-target-index                                                    [100% done]
Running wait-for-cluster-to-be-green                                           [100% done]
Running custom-vector-bulk                                                     [100% done]
Running refresh-target-index                                                   [100% done]
Running knn-query-from-data-set                                                [100% done]
------------------------------------------------------
    _______             __   _____
   / ____(_)___  ____ _/ /  / ___/_________  ________
  / /_  / / __ \/ __ `/ /   \__ \/ ___/ __ \/ ___/ _ \
 / __/ / / / / / /_/ / /   ___/ / /__/ /_/ / /  /  __/
/_/   /_/_/ /_/\__,_/_/   /____/\___/\____/_/   \___/
------------------------------------------------------
|                                                         Metric |                    Task |       Value |   Unit |
|---------------------------------------------------------------:|------------------------:|------------:|-------:|
|                     Cumulative indexing time of primary shards |                         |     3.95175 |    min |
|             Min cumulative indexing time across primary shards |                         |    0.193517 |    min |
|          Median cumulative indexing time across primary shards |                         |     1.10467 |    min |
|             Max cumulative indexing time across primary shards |                         |     2.65357 |    min |
|            Cumulative indexing throttle time of primary shards |                         |           0 |    min |
|    Min cumulative indexing throttle time across primary shards |                         |           0 |    min |
| Median cumulative indexing throttle time across primary shards |                         |           0 |    min |
|    Max cumulative indexing throttle time across primary shards |                         |           0 |    min |
|                        Cumulative merge time of primary shards |                         |           0 |    min |
|                       Cumulative merge count of primary shards |                         |           0 |        |
|                Min cumulative merge time across primary shards |                         |           0 |    min |
|             Median cumulative merge time across primary shards |                         |           0 |    min |
|                Max cumulative merge time across primary shards |                         |           0 |    min |
|               Cumulative merge throttle time of primary shards |                         |           0 |    min |
|       Min cumulative merge throttle time across primary shards |                         |           0 |    min |
|    Median cumulative merge throttle time across primary shards |                         |           0 |    min |
|       Max cumulative merge throttle time across primary shards |                         |           0 |    min |
|                      Cumulative refresh time of primary shards |                         |     19.3742 |    min |
|                     Cumulative refresh count of primary shards |                         |          34 |        |
|              Min cumulative refresh time across primary shards |                         |     5.59155 |    min |
|           Median cumulative refresh time across primary shards |                         |     6.69072 |    min |
|              Max cumulative refresh time across primary shards |                         |     7.09188 |    min |
|                        Cumulative flush time of primary shards |                         |     2.17767 |    min |
|                       Cumulative flush count of primary shards |                         |           3 |        |
|                Min cumulative flush time across primary shards |                         |     0.00215 |    min |
|             Median cumulative flush time across primary shards |                         |     0.53525 |    min |
|                Max cumulative flush time across primary shards |                         |     1.64027 |    min |
|                                        Total Young Gen GC time |                         |       0.362 |      s |
|                                       Total Young Gen GC count |                         |          21 |        |
|                                          Total Old Gen GC time |                         |           0 |      s |
|                                         Total Old Gen GC count |                         |           0 |        |
|                                                     Store size |                         |     2.87628 |     GB |
|                                                  Translog size |                         | 3.07336e-07 |     GB |
|                                         Heap used for segments |                         |           0 |     MB |
|                                       Heap used for doc values |                         |           0 |     MB |
|                                            Heap used for terms |                         |           0 |     MB |
|                                            Heap used for norms |                         |           0 |     MB |
|                                           Heap used for points |                         |           0 |     MB |
|                                    Heap used for stored fields |                         |           0 |     MB |
|                                                  Segment count |                         |          43 |        |
|                                                 Min Throughput |      custom-vector-bulk |     1889.24 | docs/s |
|                                                Mean Throughput |      custom-vector-bulk |     28058.3 | docs/s |
|                                              Median Throughput |      custom-vector-bulk |     33319.6 | docs/s |
|                                                 Max Throughput |      custom-vector-bulk |     55595.4 | docs/s |
|                                        50th percentile latency |      custom-vector-bulk |     13.1439 |     ms |
|                                        90th percentile latency |      custom-vector-bulk |     18.1077 |     ms |
|                                        99th percentile latency |      custom-vector-bulk |     69.8484 |     ms |
|                                      99.9th percentile latency |      custom-vector-bulk |     11960.3 |     ms |
|                                       100th percentile latency |      custom-vector-bulk |      240019 |     ms |
|                                   50th percentile service time |      custom-vector-bulk |     13.1439 |     ms |
|                                   90th percentile service time |      custom-vector-bulk |     18.1077 |     ms |
|                                   99th percentile service time |      custom-vector-bulk |     69.8484 |     ms |
|                                 99.9th percentile service time |      custom-vector-bulk |     11960.3 |     ms |
|                                  100th percentile service time |      custom-vector-bulk |      240019 |     ms |
|                                                     error rate |      custom-vector-bulk |        0.04 |      % |
|                                       100th percentile latency |    refresh-target-index |      240014 |     ms |
|                                  100th percentile service time |    refresh-target-index |      240014 |     ms |
|                                                     error rate |    refresh-target-index |         100 |      % |
|                                                 Min Throughput | knn-query-from-data-set |        0.01 |  ops/s |
|                                                Mean Throughput | knn-query-from-data-set |       13.04 |  ops/s |
|                                              Median Throughput | knn-query-from-data-set |       12.48 |  ops/s |
|                                                 Max Throughput | knn-query-from-data-set |       30.15 |  ops/s |
|                                        50th percentile latency | knn-query-from-data-set |     12.8036 |     ms |
|                                        90th percentile latency | knn-query-from-data-set |     15.4179 |     ms |
|                                        99th percentile latency | knn-query-from-data-set |     21.7585 |     ms |
|                                      99.9th percentile latency | knn-query-from-data-set |      397.06 |     ms |
|                                     99.99th percentile latency | knn-query-from-data-set |      155948 |     ms |
|                                       100th percentile latency | knn-query-from-data-set |      155950 |     ms |
|                                   50th percentile service time | knn-query-from-data-set |     12.8036 |     ms |
|                                   90th percentile service time | knn-query-from-data-set |     15.4179 |     ms |
|                                   99th percentile service time | knn-query-from-data-set |     21.7585 |     ms |
|                                 99.9th percentile service time | knn-query-from-data-set |      397.06 |     ms |
|                                99.99th percentile service time | knn-query-from-data-set |      155948 |     ms |
|                                  100th percentile service time | knn-query-from-data-set |      155950 |     ms |
|                                                     error rate | knn-query-from-data-set |           0 |      % |
[WARNING] Error rate is 0.04 for operation ‘custom-vector-bulk’. Please check the logs.
[WARNING] Error rate is 100.0 for operation ‘refresh-target-index’. Please check the logs.
[WARNING] No throughput metrics available for [refresh-target-index]. Likely cause: Error rate is 100.0%. Please check the logs.
---------------------------------
[INFO] SUCCESS (took 878 seconds)
---------------------------------
```
